clinica/utils/input_files.py

```diff
@@ -194,12 +199,54 @@
 # T1-Volume
 
 
+def aggregator(func):
+    """If the decorated function receives iterable arguments,
+    this decorator will call the decorated function for each
+    value in the iterable and aggregate the results in a list.
+    This works only if the iterables provided have the same length.
+    Arguments lefts as non-iterable will be repeated.
+    """
+
+    @functools.wraps(func)
+    def wrapper_aggregator(*args, **kwargs):
+        arg_sizes = [len(arg) for arg in args if isinstance(arg, Iterable)]
+        arg_sizes += [
+            len(arg) for k, arg in kwargs.items() if isinstance(arg, Iterable)
+        ]
+        if len(np.unique(arg_sizes)) > 1:
```

Comment:
Sets cost less than `numpy.unique`

```suggestion
        if len(set(arg_sizes)) > 1:
```